CPP/DesktopToastsCppApp/DesktopNotificationManagerCompat.h

```diff
@@ -0,0 +1,108 @@
+// ******************************************************************
+
+#pragma once
+#include <string>
+#include <memory>
+#include <Windows.h>
+#include <windows.ui.notifications.h>
+#include <wrl.h>
+#define TOAST_ACTIVATED_LAUNCH_ARG L"-ToastActivated"
+
+using namespace ABI::Windows::UI::Notifications;
+
+class DesktopNotificationHistoryCompat;
+
+namespace DesktopNotificationManagerCompat
+{
+    /// <summary>
+    /// If not running under the Desktop Bridge, you must call this method to register your AUMID with the Compat library and to
+    /// register your COM CLSID and EXE in LocalServer32 registry. Feel free to call this regardless, and we will no-op if running
+    /// under Desktop Bridge. Call this upon application startup, before calling any other APIs.
+    /// </summary>
+    /// <param name="aumid">An AUMID that uniquely identifies your application.</param>
+    /// <param name="clsid">The CLSID of your NotificationActivator class.</param>
+    void RegisterAumidAndComServer(const wchar_t *aumid, GUID clsid);
+
+    /// <summary>
+    /// Registers your module to handle COM activations. Call this upon application startup.
+    /// </summary>
+    void RegisterActivator();
+
+    /// <summary>
+    /// Creates a toast notifier. You must have called RegisterActivator first (and also RegisterAumidAndComServer if you're a classic Win32 app), or this will throw an exception.
+    /// </summary>
+    Microsoft::WRL::ComPtr<IToastNotifier> CreateToastNotifier();
+
+    /// <summary>
+    /// Creates a toast notification. This is simply a convenience helper method.
+    /// </summary>
+    Microsoft::WRL::ComPtr<IToastNotification> CreateToastNotification(ABI::Windows::Data::Xml::Dom::IXmlDocument* content);
+
+    /// <summary>
+    /// Creates an XmlDocument initialized with the specified string. This is simply a convenience helper method.
+    /// </summary>
+    Microsoft::WRL::ComPtr<ABI::Windows::Data::Xml::Dom::IXmlDocument> CreateXmlDocumentFromString(const wchar_t *xmlString);
+
+    /// <summary>
+    /// Gets the DesktopNotificationHistoryCompat object. You must have called RegisterActivator first (and also RegisterAumidAndComServer if you're a classic Win32 app), or this will throw an exception.
+    /// </summary>
+    std::unique_ptr<DesktopNotificationHistoryCompat> get_History();
+
+    /// <summary>
+    /// Gets a boolean representing whether http images can be used within toasts. This is true if running under Desktop Bridge.
+    /// </summary>
+    bool CanUseHttpImages();
+}
+
+class DesktopNotificationHistoryCompat
+{
+public:
+
+    /// <summary>
+    /// Removes all notifications sent by this app from action center.
+    /// </summary>
+    void Clear();
+
+    /// <summary>
+    /// Gets all notifications sent by this app that are currently still in Action Center.
+    /// </summary>
+    Microsoft::WRL::ComPtr<ABI::Windows::Foundation::Collections::IVectorView<ToastNotification*>> GetHistory();
```

Comment:
Do you need the ABI type?

Reply:
Good point, the original code was using this but seems like there's a way to not use the ABI type by looking at Desktop Bridge code samples.